src/app/server/Server.cpp

```diff
@@ -695,7 +695,7 @@ CHIP_ERROR Server::SendUserDirectedCommissioningRequest(Transport::PeerAddress c
     CHIP_ERROR err;
 
     // only populate fields left blank by the client
-    if (strlen(id.GetInstanceName()) == 0)
+    if (strnlen(id.GetInstanceName(), Dnssd::Commission::kInstanceNameMaxLength + 1) == 0)
```

Comment:
We don't care about the actual length, right?  We just care whether `*id.GetInstanceName() == '\0'`?  Does it make sense to change to that test?